lib/Fmt_ast.ml

```diff
@@ -4816,8 +4851,15 @@ and fmt_value_binding c ~rec_flag ?ext ?in_ ?epi
   let f {attr_name= {loc; _}; _} =
     Location.compare_start loc lb_exp.ast.pexp_loc < 1
   in
+  if
+    lb_pun && not (Location.contains lb_pat.ast.ppat_loc lb_exp.ast.pexp_loc)
```

Comment:
What is the `Location.contains` check here?  I would have guessed a pun was the only way for this to be true, and that it would always be true in that case.

Reply:
I added a comment to explain what's going on here.

Reply:
Does the `lb_pun` in the original `Let_binding.t` tell us whether the original binding was punned?  If so, perhaps we can just not shadow it and check that rather than this locations math?

Reply:
Good point. Fixed.

Reply:
Thanks!